rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/options/OptionEdit.vue

```diff
@@ -0,0 +1,1074 @@
+      await this.validateOption();
+      if (this.hasFormErrors) {
+        return;
+      }
+      this.$emit("update:modelValue", this.option);
+    },
+    hasError(field: string) {
+      return (
+        (this.validationErrors[field] &&
+          this.validationErrors[field].length > 0) ||
+        Object.keys(this.validationErrors).find((k) =>
+          k.startsWith(field + "."),
+        )
+      );
+    },
+    addError(field: string, error: string) {
+      if (!this.validationErrors[field]) {
+        this.validationErrors[field] = [];
+      }
+      this.validationErrors[field].push(error);
+    },
+    addWarning(field: string, error: string) {
+      if (!this.validationWarnings[field]) {
+        this.validationWarnings[field] = [];
+      }
+      this.validationWarnings[field].push(error);
+    },
+    clearValidation(field: string) {
+      delete this.validationWarnings[field];
+      delete this.validationErrors[field];
+    },
+    getProviderFor(name) {
+      return this.optionValuesPlugins.find((p) => p.name === name);
+    },
+    validateLen(field: string, max: number): boolean {
+      return !(this.option[field] && this.option[field].length > max);
+    },
+    validateRegex(field: string, regex: string): boolean {
+      let testRegex = new RegExp(regex);
+      return testRegex.test(this.option[field]);
+    },
+    validateFieldName(field: string): boolean {
+      if (Validations[field]) {
+        this.clearValidation(field);
+        return this.validateField(field, Validations[field]);
+      }
+      return true;
+    },
+    validateField(field: string, validationConfig: ValidationConfig): boolean {
+      let pass = true;
+      if (validationConfig.required) {
+        if (!this.option[field]) {
+          pass = false;
+          this.addWarning(field, this.$t("form.field.required.message"));
+        }
+      }
+      if (validationConfig.length) {
+        if (!this.validateLen(field, validationConfig.length)) {
+          pass = false;
+          this.addError(
+            field,
+            this.$t("form.field.too.long.message", {
+              max: 255,
+            }),
+          );
+        }
+      }
+      if (validationConfig.regex && this.option[field]) {
+        if (!this.validateRegex(field, validationConfig.regex)) {
+          pass = false;
+          this.addError(
+            "name",
+            this.$t(`form.option.regex.validation.error`, [
+              validationConfig.regex,
+            ]),
+          );
+        }
+      }
+      return pass;
+    },
+    validateOptionName() {
+      this.validateFieldName("name");
+    },
+    validateOptionLabel() {
+      this.validateFieldName("label");
+    },
+    localValidations() {
+      // validate all fields from Validations
+      for (const field in Validations) {
+        this.validateFieldName(field);
+      }
+    },
+    async validateOption() {
+      let res = await validateJobOption(getRundeckContext().projectName, {
```

Comment:
I will add that, I think it will be a slight improvement.  In one case it still won't catch the situation until you hit save; if you already have "required" options defined, then you change the job to scheduled=true. at that point the validation will only be performed on the server after you hit save.  I think that will be an ok tradeoff, since once the full Vue conversion is done the validation will be in sent back to the GUI before you lose your context.

Reply:
there is also a bug if the job is scheduled, the API validation isn't using that boolean as it should

Reply:
updated to fix this